src/pebble.py

```diff
@@ -402,7 +430,7 @@ def reconcile(  # noqa: C901
             ignore_string_case=True,
         )
 
-        if config_has_changed:
+        if config_has_changed or mas_config_has_changed:
```

Comment:
Does Synapse need to be restarted when MAS is restarted, or only when the MAS configuration changes?

Reply:
Synapse should only be restarted when MAS configuration changes, so even if MAS is restarted manually we don't need to restart synapse.